src/ocean_emulators/train.py

```diff
@@ -139,6 +134,20 @@ def __init__(self, cfg: TrainConfig) -> None:
         self.N_bound = len(self.boundary_var_names)
         self.N_prog = len(self.prognostic_var_names)
 
+        self.data_container = cfg.data.build(
+            data_root=cfg.experiment.resolved_data_root,
+            boundary_var_names=self.boundary_var_names,
+        )
+
+        self.mp_context: BaseContext | None
+        if cfg.data.num_workers > 0:
+            if self.data_container.supports_fork:
+                self.mp_context = multiprocessing.get_context("fork")
+            else:
+                self.mp_context = multiprocessing.get_context("spawn")
```

Comment:
🐑 A similar style preference:

```suggestion
            context_type = "fork" if self.data_container.supports_fork else "spawn"
            self.mp_context = multiprocessing.get_context(context_type)
```

What you have here is fine, too.